src/plugins/intel_gpu/src/plugin/transformations/transpose_fusion.cpp

```diff
@@ -52,13 +52,25 @@ bool has_optimized_version(const ov::Output<ov::Node>& output, bool supports_imm
     };
 
     const auto expected_dims_num = 4;
-    const auto original_dims_num = transpose_order.size();
-    if (original_dims_num < expected_dims_num) {
+
+    std::vector<size_t> order(std::begin(transpose_order), std::end(transpose_order));
+    if (expected_dims_num > order.size()) {
+        size_t orders_to_add = expected_dims_num - order.size();
+        for (size_t i = 0; i < orders_to_add; ++i)
+            order.insert(order.begin(), i);
+        for (size_t i = orders_to_add; i < order.size(); ++i)
+            order[i] = order[i] + orders_to_add;
+    }
+    auto target_permute_order = order;
+    for (size_t i = 0; i < order.size(); ++i) {
+        target_permute_order[order[i]] = i;
+    }
+    /*if (original_dims_num < expected_dims_num) {
```

Comment:
@yeonbok Hi taylor, here the permute fusion with matmul, the permute order checking logic is different from the logic inside gemm onednn(which will have potential issues, see cvs-163253),  I pull the same logic from gemm onednn to here, can you have a quick check if any concern?

Reply:
Hi @songbell  I think we should apply this change only for onednn case.